apps/challenges/views.py

```diff
@@ -2553,3 +2553,35 @@ def validate_challenge_config(request, challenge_host_team_pk):
         message = "Challenge config validation success"
         response_data = {"Success": message}
         return Response(response_data, status=status.HTTP_200_OK)
+
+
+@api_view(["GET"])
+@throttle_classes([UserRateThrottle])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((ExpiringTokenAuthentication,))
+def get_worker_logs(request, challenge_pk):
+    if not is_user_a_host_of_challenge(request.user, challenge_pk):
+        response_data = {
+            "error": "Sorry, you are not authorized to access the worker logs."
+        }
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    challenge = challenge = Challenge.objects.get(pk=challenge_pk)
+    response_data = []
+    last_submission = Submission.objects.order_by('started_at').last()
```

Comment:
What about the case when some intermediate submissions failed? In that case API won't send any logs?

Reply:
I actually need to update the code to check for submissions of only the current challenge. Will push the change in next commit.

Regarding the case where some intermediate submission fails: I remember Rishabh saying once a submission passes all the future submissions for the challenge would pass. That's the reason I was checking only for the latest submission.